src/GenericDoctrineSourceAdapter.php

```diff
@@ -45,14 +45,11 @@ public function __construct($repository, $repositoryMethod = 'findForContentMapp
     public function getObjectsOrderedById()
     {
         $entities = $this->repository->{$this->repositoryMethod}();
-        if (is_array($entities) === false) {
-            throw new \RuntimeException(
-                'The result of ' . get_class($this->repository) . '->' . $this->repositoryMethod . '() is no array, '
-                . 'which it has to be if you wish to use ' . __CLASS__
-            );
+
+        if (is_array($entities)) {
+            return new \ArrayIterator($entities);
         }
 
-        /** @var Collection mixed[] */
-        return new \ArrayIterator($entities);
+        return $entities;
```

Comment:
To honour [the return type hint in the interface](https://github.com/webfactory/content-mapping/blob/master/src/SourceAdapter.php#L22), I'd like to ensure $entities is an Iterator by now.

Reply:
added a check :+1: